src/components/svgs/nodes/qingdao-metro-num-line-badge.tsx

```diff
@@ -0,0 +1,161 @@
+import { MonoColour } from '@railmapgen/rmg-palette-resources';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { AttrsProps, CityCode } from '../../../constants/constants';
+import { MiscNodeType, Node, NodeComponentProps } from '../../../constants/nodes';
+import { AttributesWithColor, ColorField } from '../../panels/details/color-field';
+import { RmgFields, RmgFieldsField } from '@railmapgen/rmg-components';
+
+const QingdaoMetroNumLineBadge = (props: NodeComponentProps<QingdaoMetroNumLineBadgeAttributes>) => {
+    const { id, x, y, attrs, handlePointerDown, handlePointerMove, handlePointerUp } = props;
+    const {
+        num = defaultQingdaoMetroNumLineBadgeAttributes.num,
+        numEn = defaultQingdaoMetroNumLineBadgeAttributes.numEn,
+        color = defaultQingdaoMetroNumLineBadgeAttributes.color,
+        showText = defaultQingdaoMetroNumLineBadgeAttributes.showText,
+    } = attrs ?? defaultQingdaoMetroNumLineBadgeAttributes;
+
+    const onPointerDown = React.useCallback(
+        (e: React.PointerEvent<SVGElement>) => handlePointerDown(id, e),
+        [id, handlePointerDown]
+    );
+    const onPointerMove = React.useCallback(
+        (e: React.PointerEvent<SVGElement>) => handlePointerMove(id, e),
+        [id, handlePointerMove]
+    );
+    const onPointerUp = React.useCallback(
+        (e: React.PointerEvent<SVGElement>) => handlePointerUp(id, e),
+        [id, handlePointerUp]
+    );
+
+    const fgColor = color[3];
+    const [letterSpacing, sX, sY, fontSize] = Number(num) >= 10 ? [-2.4, 0, 10.25, 20] : [0, 4, 10, 22];
+
+    return (
+        <g
+            id={id}
+            transform={`translate(${x}, ${y})`}
+            onPointerDown={onPointerDown}
+            onPointerMove={onPointerMove}
+            onPointerUp={onPointerUp}
+            style={{ cursor: 'move' }}
+        >
+            <rect fill={color[2]} x="0" width="20" height="20" rx="2" ry="2" />
+            <text
+                className="rmp-name__zh"
+                textAnchor="left"
+                x={sX}
+                y={sY}
+                fill={fgColor}
+                fontSize={fontSize}
+                fontWeight="bold"
+                letterSpacing={letterSpacing}
+                dominantBaseline="central"
+            >
+                {num}
+            </text>
+            {showText && (
+                <>
+                    <text className="rmp-name__zh" x="22" y="10.5" fontSize="13">
+                        号线
+                    </text>
+                    <text className="rmp-name__en" x="22.5" y="19.5" fontSize="8">
+                        Line {numEn}
+                    </text>
+                </>
+            )}
+        </g>
+    );
+};
+
+/**
+ * Qingdao Metro Num Line Badge specific props.
+ */
+export interface QingdaoMetroNumLineBadgeAttributes extends AttributesWithColor {
+    num: number;
+    numEn: number | string;
+    showText: boolean;
```

Comment:
What's the real world image of a badge with out text? It would be great to update the issue with this additional info :)

Reply:
<img width="519" alt="image" src="https://github.com/railmapgen/rmp/assets/59787082/26782247-6347-4845-9b2c-7d1c419057d2">

There's a special case. numEn = `Line 3, 8`, showText = `false`